pkg/ebpf/signature_engine.go

```diff
@@ -45,6 +46,35 @@ func (t *Tracee) engineEvents(ctx context.Context, in <-chan *trace.Event) (<-ch
 
 	go t.sigEngine.Start(ctx)
 
+	// Create a function for feeding the engine with an event
+	feedFunc := func(event *trace.Event) {
```

Comment:
Nice refactor. Since this file is already made in particular for the engine related stuff, do you mind making this a top level function in the file? See if it looks better, if it doesn't no need.

Reply:
Hmm I don't think it will look better as it uses the channels of this stage.
So unless we pass to it all channels each time, or create a function that return this function (which looks the same to me like the current state or even worst) I don't think we can improve the function here.